packages/common/src/audius-query/createApi.ts

```diff
@@ -553,6 +557,130 @@ const buildEndpointHooks = <
+    const [status, setStatus] = useState<Status>(Status.IDLE)
+
+    // A soft reset resets our current page and loading state, but the data is still held in store (which means cache hits will still return as expected)
+    // This is triggered below when the args or page size changes
+    const softReset = useCallback(() => {
+      setPage(0)
+      setLoadingMore(false)
+    }, [])
+
+    // Includes softReset above BUT also hard resets cached data in store; also resets loading status
+    // Can be used to hard refresh all data and reset the page
+    const hardReset = useCallback(() => {
+      softReset()
+      setStatus(Status.LOADING)
+      // @ts-ignore - Unclear whats wrong with they type here
+      dispatch(actions[`reset${capitalize(endpointName)}`]())
+    }, [dispatch, softReset])
+
+    // Only need to soft reset when args or pagesize changes - leaves all cached data intact
+    useCustomCompareEffect(softReset, [baseArgs], isEqual)
+
+    const args = {
+      ...baseArgs,
+      limit: pageSize,
+      offset: page * pageSize
+    } as ArgsType & PaginatedQueryArgs
+
+    const allPageData = useSelector((state: CommonState) => {
+      // If in single page mode we only return the current page of data (no aggregation)
+      if (singlePageData) {
+        return state?.api?.[reducerPath]?.[endpointName]?.[
+          getKeyFromFetchArgs(args)
+        ]?.normalizedData
+      }
+
+      // By default, select data for every page up to the current page and aggregate the results
+      const pageAccumulator = []
+
+      for (let i = 0; i <= page; i++) {
+        const key = getKeyFromFetchArgs({
+          ...args,
+          offset: i * pageSize
+        })
+        const normalizedPageData =
+          state?.api?.[reducerPath]?.[endpointName]?.[key]?.normalizedData
+        if (normalizedPageData) {
+          pageAccumulator.push(...normalizedPageData)
+        }
+      }
+      return pageAccumulator
+    })
+
+    // Since this is a paginated query we know that the results will be in an array
+    // Call the useQuery hook to trigger cache checks and/or fetches
+    const result = useQuery(args, queryHookOptions) as QueryHookResults<Data[]>
+
+    // If forceLoad is set, force a data refresh on every render
+    useEffect(() => {
+      if (forceLoad) {
+        result.forceRefresh()
+        setForceLoad(false)
+      }
+    }, [result, forceLoad])
+
+    // Track status of the latest page fetches in order to update loadingMore
+    useCustomCompareEffect(
+      () => {
+        setStatus(result.status)
+        if (result.status === Status.ERROR) {
+          setLoadingMore(false)
+          return
+        }
+        if (result.status === Status.SUCCESS) {
+          setLoadingMore(false)
+        }
+      },
+      [result.status, args],
+      isEqual
+    )
+
+    const notError = result.status !== Status.ERROR
+    const stillLoadingCurrentPage =
+      loadingMore || result.status === Status.LOADING
```

Comment:
I think this `loadingMore` is what that comment was mentioning about having another `RELOADING` status.

I think local state is probably fine here, but we could also add a state to the enum